### PR TITLE
DIS-217 Test for trending score for new threads

### PR DESCRIPTION
**Pull request by armonr**
It'd be ideal to just have a sproc for the trending formula, but since I cannot think of a use case other than this test, an approximation of it in the integration test should suffice.

@jimmordino @jogura

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 39 issues:
* MAJOR 21 major
* MINOR 18 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L161) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L71) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L86) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Method has 8 parameters, which is greater than 7 authorized.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L442) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L333) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L76) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L66) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L389) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L78) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L242) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L82) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L74) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L205) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Remove useless curly braces around statement](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L130) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1602)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Define a constant instead of duplicating this literal "siteId" 3 times.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L447) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L311) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L400) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Method has 8 parameters, which is greater than 7 authorized.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L510) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L305) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L68) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L361) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L137) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Method has 8 parameters, which is greater than 7 authorized.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L491) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L89) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L395) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L232) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L96) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L384) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L178) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L216) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/e915e5778d4b45c4eeb509fa9087bfd5adae792e/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L166) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java [hunk 1]
```diff
@@ -345,6 +348,8 @@ public void createEmptyThreadOnTestForumExpectSuccess() throws Exception {
     ForumThread retrieveCreatedThread =
         getThreadById(SITE_ID_TEST, createdThread.getId(), ResponseGroup.FULL);
 
+    assertThat(retrieveCreatedThread.getTrendingScore(),
+               Matchers.greaterThan(getTrendingScoreMinusOnePercent(createdThread)));
```
Comment (jimmordino) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java]: Can you use a ```closeTo``` matcher and not worry about subtracting one percent?
```
static Matcher<java.lang.Double>	closeTo(double operand, double error) 
          Creates a matcher of Doubles that matches when an examined double is equal to the specified operand, 
within a range of +/- error.
```

Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java]: I did not know this existed! Will fix.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 39 issues:
* MAJOR 21 major
* MINOR 18 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Method has 8 parameters, which is greater than 7 authorized.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L490) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L178) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L388) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L89) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L360) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L399) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Method has 8 parameters, which is greater than 7 authorized.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L509) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L96) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L161) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L305) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L216) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L66) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Method has 8 parameters, which is greater than 7 authorized.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L441) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L242) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L394) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L166) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L205) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L71) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L68) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L311) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L383) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L74) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L78) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L137) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L333) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L82) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Remove useless curly braces around statement](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L130) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1602)
* MAJOR [Rename "userId" which hides the field declared at line 82.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L232) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L86) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Define a constant instead of duplicating this literal "siteId" 3 times.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L446) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/1d41220557cc296406bd06b43dc9812ef40b7a05/service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceIntegrationTest.java#L76) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

**Pull request merged**